### PR TITLE
Add a test for object identity in WebAssembly exports.

### PR DESCRIPTION
**Pull request by Ms2ger**

### CODE REVIEW

**Comment by Ms2ger** [comment]
CC @lukewagner

File: wasm/jsapi/instance/constructor-caching.any.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+  const buffer = builder.toBuffer();
+
+  const module = new WebAssembly.Module(buffer);
+  const instance = new WebAssembly.Instance(module, {
+    "module": {
+      "imported": exports.fn,
+      "global": exports.global,
+      "memory": exports.memory,
+      "table": exports.table,
```
Comment (xtuc) [inline wasm/jsapi/instance/constructor-caching.any.js]: You could do `{ module: exports }` as well, but might make it harder to understand

File: wasm/jsapi/instance/constructor-caching.any.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    .addFunction("fn", kSig_v_d)
+    .addBody([])
+    .exportFunc();
+
+  builder.setTableLength(1);
+  builder.addExportOfKind("table", kExternalTable, 0);
+
+  builder.addGlobal(kWasmI32, false)
+    .exportAs("global")
+    .init = 7;
```
Comment (xtuc) [inline wasm/jsapi/instance/constructor-caching.any.js]: The value isn't used, could we remove the assignement?

**Review by xtuc — APPROVED** [review]
otherwise lgtm

File: wasm/jsapi/instance/constructor-caching.any.js (hunk 2 above)
Comment (Ms2ger) [reply wasm/jsapi/instance/constructor-caching.any.js]: Sure.

File: wasm/jsapi/instance/constructor-caching.any.js (hunk 1 above)
Comment (Ms2ger) [reply wasm/jsapi/instance/constructor-caching.any.js]: I think it would be too opaque, yeah.

**Pull request merged**